packages/SwingSet/src/kernel/virtualObjectManager.js

```diff
@@ -253,8 +263,8 @@ export function makeVirtualObjectManager(
     function virtualObjectKey(key) {
       const vobjID = valToSlot.get(key);
       if (vobjID) {
-        const { type, virtual } = parseVatSlot(vobjID);
-        if (type === 'object' && virtual) {
+        const { type, virtual, allocatedByVat } = parseVatSlot(vobjID);
```

Comment:
This looks great for addressing #3115: keys which are Presences or Representatives use their vref as the index, keys which are Remotables (or any non-vref-bearing object) use the `Object` identity.